### ISSUE TITLE
dependencies are incompatible with Phoenix

### ISSUE THREAD

**Issue by kagux** [stephenmoloney/fastimage#3]
I've spend a few hours trying to reconcile dependencies. The issue is coming from `gun` requiring `cowlib 1.3.0`, while Phoenix is on `cowboy ~> 1.0.0` which requires `cowlib ~> 1.0.0` and is incompatible if you try to force `1.3.0` on it. 
With lower `gun` version that works with `cowlib ~> 1.0.0` your lib breaks.
With `cowlib ~> 1.0.0, override: true` `gun` compilation break.
In the end I got it to work with my `gun` fork that relaxes `cowlib` dependency.

```
     {:cowlib, "~> 1.0.0", manager: :rebar, override: true},
     {:gun, github: "kagux/gun", override: true},
     {:ranch, "~> 1.2.0", manager: :rebar, override: true},
```

But I can't be sure of all consequences. Could be that issue I reported before is related to this change.
I think it would benefit your lib a lot to be compatible with latest version of Phoenix.

**Comment by stephenmoloney** [comment]
- I agree with you that these dependencies issues are not ideal.
- Unfortunately, the only version of gun available is [1.0.0-pre.1](https://hex.pm/packages/gun/1.0.0-pre.1) and I've no idea when that library might be updated. 

Potential solution:
- I'm half thinking of changing to another http client, :hackney maybe. I'll try to look in to this issue today and the other issue.

**Comment by stephenmoloney** [comment]
This should be resolved now. I swapped the client from `:gun` to `:hackney` version 1.6 which is the one used by latest `httpoison`. 
This also solves the [PR](https://github.com/stephenmoloney/fastimage/pull/1) as a side-effect. 

Thanks for this.
